### ISSUE TITLE
Redes Neurais Profundas Triplet Aplicadas à Classificação de Sinais em Interfaces Cérebro-Computador

### ISSUE THREAD

**Issue by Kotzly** [Kotzly/BCI_MsC#4]
Review of [Redes Neurais Profundas Triplet Aplicadas à Classificação de Sinais em Interfaces Cérebro-Computador](https://github.com/Kotzly/BCI_MsC/files/6105543/Redes_Neurais_Profundas_Triplet_Aplicadas_a_Classificacao_de_Sinais_em_Interfaces_Cerebro-Computador.pdf).

Pedro R. A. S. Bassi*, Willian Rampazzo and Romis Attux

**Comment by Kotzly** [comment]
## Abstract
> This work proposes an approach to EEG signal classification in brain-computer interfaces based on triplet deep neural networks. The network is tested over a database with ten users, in a setup that excludes the evaluated user’s data from the network training process. The results are promising, although further investigation is necessary to verify if there are additional nonlinear relationships in the data to be explored.

**Comment by Kotzly** [comment]
## Dataset

The dataset was collected by the authors. It consists of 10 healthy subjects, and the stimulus was two 3,8cm chess pattern squares flickering at 12 and 15Hz. Each subject participate in 8 sessions of 12 seconds for each frequency. The EEG was captured from 16 electrodes arranged in the 10-10 international pattern using a sampling rate of 256Hz.

![image](https://user-images.githubusercontent.com/22718719/110727386-edb78800-81f9-11eb-8fe2-046484bb4c26.png)

## Methods

The trainining was made using the triplet loss function with an alpha of 0,8. The anchor, negative example and positive example were chosen randomly, but all images from the training set are used as anchors once.

![image](https://user-images.githubusercontent.com/22718719/110727685-7df5cd00-81fa-11eb-989b-8f19dbb39389.png)


The 12 second windows for each session were divided in 3 second windows with 1 second stride. Each sliced window is transformed to a spectogram using the STFT, and the STFT is converted to decibels. The frequencies used after the STFT are the principal and the first higher harmonics of the target frequencies, and also 1 Hz higher and lower (11, 12, 13, 23, 24, 25, 29, 30, 31Hz).

The proposed network is as follows:

![image](https://user-images.githubusercontent.com/22718719/110728498-f610c280-81fb-11eb-9aaf-273539eebb63.png)

## Results

The results are as follows:

![image](https://user-images.githubusercontent.com/22718719/110728586-1ccef900-81fc-11eb-9036-049ba982694f.png)

The results showed that every method performed similarly, with the linear classification on the spectogram being the better on average (but not significantly better).
